codec/amino_codec.go

```diff
@@ -1,18 +1,25 @@
 package codec
 
 import (
+	"fmt"
+
 	"github.com/cosmos/gogoproto/proto"
+	protov2 "google.golang.org/protobuf/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
 )
 
-// AminoCodec defines a codec that utilizes Codec for both binary and JSON
-// encoding.
+// Deprecated: AminoCodec defines a codec that utilizes Codec for both binary and JSON
```

Comment:
its unclear how this is deprecated without supporting JSONcodec, is there a way you envision to manage this?